code/modules/client/asset_cache.dm

```diff
@@ -482,117 +482,54 @@ var/list/asset_datums = list()
-	if(!istype(v_type))
-		return
-	for(var/products in list(v_type?.products, v_type?.contraband, v_type?.premium))
-		for(var/o_path in products)
-			var/obj/O = new o_path
-			var/i_type = ckey("[O.type]")
-			var/icon/I = icon(O.icon, O.icon_state)
-			if(istype(O, /obj/item/seeds))
-				// thanks seeds for being overlays defined at runtime
-				var/obj/item/seeds/S = O
-				I = S.update_appearance(TRUE)
-				Insert(i_type, I, forced=I)
-			else
-				if(i_type in sprites)
-					continue
-				if(O.overlay_queued)
-					O.compile_overlays()
-				if(O.overlays.len)
-					I = getFlatIcon(O)
-					Insert(i_type, I, forced=I)
-				else
-					Insert(i_type, O.icon, O.icon_state)
-
-	..()
-
-// v_type needs to be the path to the vending machine
-
-/datum/asset/spritesheet/vending/vendors
-	v_type = /obj/machinery/vending/vendors
-
-/datum/asset/spritesheet/vending/boozeomat
-	v_type = /obj/machinery/vending/boozeomat
-
-/datum/asset/spritesheet/vending/assist
-	v_type = /obj/machinery/vending/assist
-
-/datum/asset/spritesheet/vending/coffee
-	v_type = /obj/machinery/vending/coffee
-
-/datum/asset/spritesheet/vending/snack
-	v_type = /obj/machinery/vending/snack
-
-/datum/asset/spritesheet/vending/cola
-	v_type = /obj/machinery/vending/cola
-
-/datum/asset/spritesheet/vending/cigarette
-	v_type = /obj/machinery/vending/cigarette
-
-/datum/asset/spritesheet/vending/medical
-	v_type = /obj/machinery/vending/medical
-
-/datum/asset/spritesheet/vending/phoronresearch
-	v_type = /obj/machinery/vending/phoronresearch
-
-/datum/asset/spritesheet/vending/wallmed1
-	v_type = /obj/machinery/vending/wallmed1
-
-/datum/asset/spritesheet/vending/wallmed2
-	v_type = /obj/machinery/vending/wallmed2
-
-/datum/asset/spritesheet/vending/security
-	v_type = /obj/machinery/vending/security
-
-/datum/asset/spritesheet/vending/hydronutrients
-	v_type = /obj/machinery/vending/hydronutrients
-
-/datum/asset/spritesheet/vending/hydroseeds
-	v_type = /obj/machinery/vending/hydroseeds
-
-/datum/asset/spritesheet/vending/magivend
-	v_type = /obj/machinery/vending/magivend
-
-/datum/asset/spritesheet/vending/dinnerware
-	v_type = /obj/machinery/vending/dinnerware
-
-/datum/asset/spritesheet/vending/sovietsoda
-	v_type = /obj/machinery/vending/sovietsoda
-
-/datum/asset/spritesheet/vending/tool
-	v_type = /obj/machinery/vending/tool
-
-/datum/asset/spritesheet/vending/engivend
-	v_type = /obj/machinery/vending/engivend
-
-/datum/asset/spritesheet/vending/tacticool
-	v_type = /obj/machinery/vending/tacticool
-
-/datum/asset/spritesheet/vending/tacticoolert
-	v_type = /obj/machinery/vending/tacticool/ert
-
-/datum/asset/spritesheet/vending/engineering
-	v_type = /obj/machinery/vending/engineering
+	var/list/vending_products = list()
+	for(var/v_type in typesof(/obj/machinery/vending))
+		var/obj/machinery/vending/V = new v_type
+		if(!istype(V))
+			continue
+		for(var/list/p in list(V?.products, V?.contraband, V?.premium))
+			for(var/k in p)
+				vending_products += k
```

Comment:
All of these typechecks are unnecessary. E.g: there's absolutely no way you're getting a different type than what you asked for in a `typesof()` loop. Similiarly we don't need to check if a vending machine exists because the type will always be valid.